mmcv/runner/base_runner.py

```diff
@@ -49,14 +53,15 @@ class BaseRunner(metaclass=ABCMeta):
     """
 
     def __init__(self,
-                 model,
-                 batch_processor=None,
-                 optimizer=None,
-                 work_dir=None,
-                 logger=None,
-                 meta=None,
-                 max_iters=None,
-                 max_epochs=None):
+                 model: torch.nn.Module,
+                 batch_processor: Optional[Callable] = None,
+                 optimizer: Optional[Union[Dict,
+                                           torch.optim.Optimizer]] = None,
```

Comment:
For multiple types, use Union[..., None] instead of Optional[Union[...]]
Refers to [Google Style Book](https://github.com/google/styleguide/blob/gh-pages/pyguide.md)

Reply:
@teamwong111 thanks! cc @HAOCHENYE it seems that we'd better use `Union` instead of `Optional` with more than one input argument

![image](https://user-images.githubusercontent.com/12531708/173395557-1aae2b47-7234-4940-80d9-3125cd1f9708.png)